src/Command/AnalyseApplication.php

```diff
@@ -58,7 +58,13 @@ public function analyse(
 				$errorOutput->writeLineFormatted('Result cache was not saved because of ignoredErrorHelperResult errors.');
 			}
 		} else {
-			$resultCache = $resultCacheManager->restore($files, $debug, $onlyFiles, $projectConfigArray, $errorOutput);
+			$resultCacheName = null;
```

Comment:
while thinking about the idea suggested in https://github.com/phpstan/phpstan/issues/7379#issuecomment-1164150916 I came to this - in my eyes  - even simpler solution to the problem.

instead of fiddling with all the different path-types involved we just use a separate resultCache-name per project-configuration file.

tbh the solutions seems to be to simple, so I think it might have some serious problems somewhere ;-).

in case we can proceed with this idea, I would check the current build errors and see how to actually test this change...

any feedback is welcome.

Reply:
The `$resultCacheName` parameter of `restore` method shouldn't be misused for this, its usage is reserved only for PHPStan Pro purposes. It puts the result cache file in a subdirectory, I don't want to complicate it like that.

But hey - the logic you're doing here depends on `$projectConfigArray` and that is entirely available in `restore()` method too, so you can do the logic based on that there. A lot of useful values that can be used in the hash computation logic are already in the constructor/can be asked for there so I think this idea has wings.

Reply:
adjusted per feedack and manually tested locally.

do we need a automatic test for the case?

Reply:
btw: in the docs there is no `resultCachePath` mentioned.. should it be added?

https://phpstan.org/config-reference#caching